owners/src/owners_check.js

```diff
@@ -59,53 +60,100 @@ class OwnersCheck {
   /**
    * Constructor.
    *
+   * @param {!LocalRepository} repo local repository to read from.
    * @param {!GitHub} github GitHub API interface.
    * @param {!PullRequest} pr pull request to run owners check on.
    */
-  constructor(github, pr) {
-    Object.assign(this, {github, pr});
+  constructor(repo, github, pr) {
+    const parser = new OwnersParser(repo, github.log);
+
+    Object.assign(this, {github, pr, repo, parser});
+
+    this.tree = null;
+    this.approvers = null;
+    this.changedFiles = null;
+    this.initialized = false;
+  }
+
+  /**
+   * Initializes key properties requiring async/await.
+   */
+  async init() {
+    await this.repo.checkout();
+    this.tree = await this.parser.parseOwnersTree();
+    this.approvers = await this.getApprovers();
+    this.changedFiles = await this.github.listFiles(this.pr.number);
+    this.initialized = true;
+  }
+
+  /**
+   * Identifies all reviewers whose latest reviews are approvals.
+   *
+   * Also includes the author, unless the author has explicitly left a blocking
+   * review.
+   *
+   * @return {string[]} list of usernames.
+   */
+  async getApprovers() {
```

Comment:
Looks like this should be a private function, and external callers should get `this.approvers` instead

Reply:
Done.